.github/workflows/release-nuget.yml

```diff
@@ -0,0 +1,34 @@
+# Workflow creating a Github package
+name: Release Nuget
+
+on:
+  push:
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Verify commit exists in origin/main
+      run: |
+        git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
+        git branch --remote --contains | grep origin/main
+    - name: Set VERSION variable from tag
+      run: echo "VERSION=${GITHUB_REF/refs\/tags\/v/}" >> $GITHUB_ENV
+    - name: Set SOLUTION variable to point to solution file
+      run: |
+        SOLUTION=$(find . -name '*.sln' -printf "%p" -quit)
+        echo "SOLUTION=${SOLUTION}" >> $GITHUB_ENV
+    - name: Build
+      run: dotnet build --configuration Release /p:Version=${VERSION} ${SOLUTION}
+    - name: Test
+      run: dotnet test --configuration Release /p:Version=${VERSION} --no-build ${SOLUTION}
+    - name: Pack
+      run: dotnet pack --configuration Release /p:Version=${VERSION} --no-build --output . ${SOLUTION}
+    - name: Push
+      run: dotnet nuget push *.${VERSION}.nupkg --source https://nuget.pkg.github.com/${GITHUB_ACTOR}/index.json --api-key ${GITHUB_TOKEN}
```

Comment:
Btw, why are we pushing to github.com instead of nuget.org?

Reply:
As I understand @altinnadmin, the idea was to avoid having to place nuget.org (or npmjs.org, mavencentral.org or whatever) credentials into each repository settings, but control this centrally. But it's not clear to me how this should be implemented - from what I can tell this will require integrating the Github API in some way.

Another aspect was to avoid polluting nuget.org et. al. with packages that are pre-release/internal in nature. But this is not really an issue with this workflow, where one has to actually create a release in order for a package to be published. We can easily create a separate workflow triggered upon merge to main that creates a pre-release package that doesn't have to be pushed anywhere.

Reply:
Pre-release packages (that eventually are supposed to go out in public) and that have the pre-release part of the version number would be hidden unless you ask for them. Don't see that as a big problem, but not having a problem with them not being published either. But if we are supposed to use these packages on our development environment we would need to set up another nuget source. 

See the point in not having the credentials in every git repository, at least if it's the same. Perhaps we should have an explicit set of credentials for every repository?

Other than that I think this is a good start, but I'll leave it up to @altinnadmin to approve as he has the bigger picture.

Reply:
> as he has the bigger picture

The idea was to test publishing of packages to github IF that could simplify/improve anything for us or externals, f.ex. for pre-relase or internal packages or handling of secrets. 

If there are no positive effects then we should reconsider.

> Perhaps we should have an explicit set of credentials for every repository

Yeah, ref. our wish for teams with ownership of their owen repos/products, that makes sense.

Reply:
@altinnadmin @RonnyB71 
I've added another workflow that runs upon push to main, which generates a package version based on the latest tag (if it exists). For instance if the latest tag was "v2.34.56", the version will be set to "2.34.56-next.20211118.8" where the two latter parts are the current date and $GIT_RUN_NUMBER, making a unique and incrementing/sortable semver 2.0-compliant version. 

I believe this will cover any internal pre-release needs, making packages available automatically whenever a PR is merged to main. For "official" releases, I agree it makes sense to use the first workflow and push to nuget.org directly by creating a tag/release using per-repo credentials. I've changed release-nuget.yaml to reflect this, it now requires a secret being defined in the repo called NUGET_ORG_API_KEY.

Reply:
Nice job - I have no further questions:)

Reply:
@altinnadmin Are we good to go here? We can merge this first, and configure nuget.org-keys.

Reply:
@elsand LGTM 👍